apps/api/src/domain/games/memory-match/queries/memory-match-admin.unpublish.e2e.spec.ts

```diff
@@ -0,0 +1,106 @@
+import { MemoryMatchModule } from '../memory-match.module';
+import {
+    IMemoryMatchRepository,
+    MEMORY_MATCH_REPOSITORY_INJECTION_TOKEN,
+} from '../memory-match.repository.interface';
+import { MemoryMatchRound } from '../models/memory-match-round.entity';
+
+const buildEndpoint = (id: string) => `games/memory-match/${id}/unpublish`;
+
+const testMemoryRoundId = buildDummyUuid(432);
+
+const publishedMemoryMatchRound = buildTestInstance(MemoryMatchRound, {
+    id: testMemoryRoundId,
+    isPublished: true,
+});
+
+describe(`when using the REST API to unpublish a memory match round`, () => {
+    let app: INestApplication;
+
+    let memoryMatchRepository: IMemoryMatchRepository;
+
+    const setItUp = async (user?: CoscradUserWithGroups) => {
+        const testModule = await Test.createTestingModule({
+            imports: [
+                ConfigModule.forRoot({
+                    isGlobal: true,
+                    envFilePath: buildConfigFilePath(Environment.test),
+                    cache: false,
+                }),
+                PersistenceModule.forRootAsync(),
+                MemoryMatchModule,
+            ],
+        })
+            .overrideGuard(AdminJwtGuard)
+            .useValue(new MockJwtAdminAuthGuard(user))
+            .overrideProvider(ConfigService)
+            .useValue(
+                buildMockConfigService({
+                    ARANGO_DB_NAME: generateDatabaseNameForTestSuite(),
+                })
+            )
+            .compile();
+
+        app = testModule.createNestApplication();
+
+        await app.init();
+
+        memoryMatchRepository = app.get(MEMORY_MATCH_REPOSITORY_INJECTION_TOKEN);
+    };
+
+    describe(`when the user is a COSCRAD admin`, () => {
+        const coscradAdminUser = buildTestInstance(CoscradUser, {
+            roles: [CoscradUserRole.superAdmin],
+        });
+
+        beforeAll(async () => {
+            await setItUp(new CoscradUserWithGroups(coscradAdminUser, []));
+        });
+
+        beforeEach(async () => {
+            await app
+                .get(ArangoDatabaseProvider)
+                .getDatabaseForCollection('memory-match-rounds')
+                .clear();
+        });
+
+        afterAll(async () => {
+            await app.close();
+
+            app.get(ArangoDatabaseProvider).close;
+        });
+
+        describe(`when there is an existing round`, () => {
+            describe(`when the round is published`, () => {
+                beforeEach(async () => {
+                    await memoryMatchRepository.create(publishedMemoryMatchRound);
+                });
+
+                it(`should unpublish the round`, async () => {
+                    const res = await request(app.getHttpServer()).patch(
+                        buildEndpoint(testMemoryRoundId)
+                    );
+
+                    expect(res.status).toBe(HttpStatusCode.ok);
```

Comment:
Use the repository to fetch the `updatedRound` and `expect(updatedRound.isPublished).toBe(false)`